framework/src/ble_manager/ble_manager_state.c

```diff
@@ -197,6 +267,137 @@ ble_result_e blemgr_handle_request(blemgr_msg_s *msg)
 		ret = ble_drv_conn_is_any_active(is_active);
 	} break;
 
+	// Scanner
+	case BLE_EVT_CMD_START_SCAN: {
+		BLE_STATE_CHECK;
+
+		if (g_scan_ctx.state != BLE_SCAN_STOPPED) {
+			ret = TRBLE_BUSY;
+			break;
+		}
+
+		ble_scan_state_e priv_state = g_scan_ctx.state;
+		blemgr_msg_params *param = (blemgr_msg_params *)msg->param;
+		ble_scan_filter *filter = (ble_scan_filter *)param->param[0];
+		ble_scan_callback_list *callbacks = (ble_scan_callback_list *)param->param[1];
+
+		if (callbacks == NULL) {
+			ret = TRBLE_INVALID_ARGS;
+			break;
+		}
+		memcpy(&(g_scan_ctx.callback), callbacks, sizeof(ble_scan_callback_list));
+
+		if (filter != NULL) {
+			memcpy(&(g_scan_ctx.filter), filter, sizeof(ble_scan_filter));
+		}
+
+		g_scan_ctx.state = BLE_SCAN_CHANGING;
+		ret = ble_drv_start_scan((trble_scan_filter *)filter);
+		if (ret != TRBLE_SUCCESS) {
+			memset(&(g_scan_ctx.callback), 0, sizeof(ble_scan_callback_list));
+			memset(&(g_scan_ctx.filter), 0, sizeof(ble_scan_filter));
+			g_scan_ctx.state = priv_state;
+		}
+	} break;
+
+	case BLE_EVT_CMD_STOP_SCAN: {
+		BLE_STATE_CHECK;
+
+		ret = TRBLE_SUCCESS;
+		if (g_scan_ctx.state != BLE_SCAN_STARTED) {
+			break;
```

Comment:
In this case, `ret` will be `TRBLE_SUCCESS`.
But I think this is failure case.
Or is it intended to return SUCCESS if trying to stop the already stopped?
Please check this.